Rectangular linear_map from HPolygon crashes

```julia
m = 4
X = convert(HPolygon, rand(BallInf))
A = rand(m, 2)
Y = linear_map(A, X)

ERROR: AssertionError: comparison of vectors `u` and `v` by their direction requires they are of length 2, but their lengths are 4 and 4 respectively
Stacktrace:
 [1] macro expansion at .julia/dev/LazySets/src/Assertions/Assertions.jl:23 [inlined]
 [2] <=(::Array{Float64,1}, ::Array{Float64,1}) at .julia/dev/LazySets/src/Interfaces/AbstractPolygon.jl:143
 [3] addconstraint!(::Array{LazySets.HalfSpace{Float64,Array{Float64,1}},1}, ::LazySets.HalfSpace{Float64,Array{Float64,1}}; linear_search::Bool, prune::Bool) at .julia/dev/LazySets/src/Interfaces/AbstractHPolygon.jl:448
 [4] HPolygon(::Array{LazySets.HalfSpace{Float64,Array{Float64,1}},1}; sort_constraints::Bool, check_boundedness::Bool, prune::Bool) at .julia/dev/LazySets/src/Sets/HPolygon.jl:56
 [5] HPolygon at .julia/dev/LazySets/src/Sets/HPolygon.jl:52 [inlined]
 [6] _linear_map_hrep_helper at .julia/dev/LazySets/src/Interfaces/AbstractHPolygon.jl:597 [inlined]
 [7] linear_map(::Array{Float64,2}, ::HPolygon{Float64,LazySets.Arrays.SingleEntryVector{Float64}}; algorithm::Nothing, check_invertibility::Bool, cond_tol::Float64, inverse::Nothing, backend::Nothing, elimination_method::Nothing) at .julia/dev/LazySets/src/Interfaces/AbstractPolyhedron_functions.jl:621
 [8] linear_map(::Array{Float64,2}, ::HPolygon{Float64,LazySets.Arrays.SingleEntryVector{Float64}}) at .julia/dev/LazySets/src/Interfaces/AbstractPolyhedron_functions.jl:592
 [9] top-level scope at REPL[247]:1
 [10] eval(::Module, ::Any) at ./boot.jl:331
 [11] eval_user_input(::Any, ::REPL.REPLBackend) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [12] run_backend(::REPL.REPLBackend) at .julia/packages/Revise/XFtoQ/src/Revise.jl:1162
 [13] top-level scope at none:0
```

--- a/src/Interfaces/AbstractHPolygon.jl
+++ b/src/Interfaces/AbstractHPolygon.jl
@@ -597,10 +597,3 @@ function isbounded(P::AbstractHPolygon, use_type_assumption::Bool=true)
     end
     return isbounded_unit_dimensions(P)
 end
-
-function _linear_map_hrep_helper(M::AbstractMatrix{N}, P::AbstractHPolygon{N},
-            algo::AbstractLinearMapAlgorithm) where {N<:Real}
-    constraints = _linear_map_hrep(M, P, algo)
-    T = basetype(P)
-    return T(constraints)
-end

--- a/test/unit_Polygon.jl
+++ b/test/unit_Polygon.jl
@@ -248,9 +248,15 @@ for N in [Float64, Float32, Rational{Int}]
     @test intersection(p3, p4) isa EmptySet{N}
 
     # concrete linear map
-    # in 2D and for an invertible map we get an HPolygon; see #631 and #1093
+    # in 2D and for an invertible map we get an HPolygon (see #631 and #1093)
     HP = convert(HPolygon, BallInf(N[0, 0], N(1)))
     @test linear_map(N[1 0; 0 2], HP) isa HPolygon{N}
+    # in higher dimensions we get an HPolytope (#2168)
+    if test_suite_polyhedra
+        p4 = convert(HPolygon, BallInf(zeros(N, 2), N(1)))
+        A = ones(N, 4, 2)
+        @test linear_map(A, p4) isa HPolytope
+    end
 
     # vertices_list removes duplicates by default (#1405)
     p3 = HPolygon([HalfSpace(N[1, 0], N(0)), HalfSpace(N[0, 1], N(0)),

